CHANGELOG.next.asciidoc

```diff
@@ -46,6 +46,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Added `container.image.name` to `journald` Filebeat input's Docker-specific translated fields. {pull}40450[40450]
 - Change log.file.path field in awscloudwatch input to nested object. {pull}41099[41099]
 - Remove deprecated awscloudwatch field from Filebeat. {pull}41089[41089]
+- `max_number_of_messages` config for S3 input's SQS mode is now ignored. Instead use `number_of_workers` to scale ingestion rate in both S3 and SQS modes. {pull}40699[40699]
```

Comment:
What impact will this have for users that were relying on `max_number_of_messages` to try to tune the input?

Is it just going to perform better with no user intervention? Could there be unintended side effects, like agents running out of memory because they can now fill up their queues?

Under what circumstances would `number_of_workers` be something a user has to adjust?

Reply:
> What impact will this have for users that were relying on max_number_of_messages to try to tune the input?

For users that set the field to something unreasonably high to improve throughput (which was previously the only way to work around this bottleneck), they will instead get good performance by default with no added cost. The only scenario that should be "bad" in this change is if the input objects are very small (the worst-case bottleneck) _and_ they _intentionally_ tuned max_number_of_messages low so that ingestion would be extremely slow. In that case, ingestion speed will significantly increase (which costs a fair amount more network bandwidth, and marginally more CPU). [Though "I used this special config value to intentionally throttle my ingestion to one event per second" is very much in the "holding the escape key down to warm up the room" category imo -- _total_ ingestion costs should not go up, it should just empty the initial queue faster.]

> agents running out of memory because they can now fill up their queues?

It isn't _impossible_ for this to happen -- if users configured a queue larger than they actually used, and were relying on the fact that this input never has more than 5-10 active events, then the new version will use more memory. However, even a _full_ queue is usually less than 10% of Filebeat's memory footprint, so this wouldn't be a significant factor unless they had explicitly configured a _very_ large queue that they never use.

> Under what circumstances would number_of_workers be something a user has to adjust?

If they have a cap on network resources, or to a lesser extent CPU ("number of workers" is also "number of potentially parallel AWS API calls/downloads", and also number of downloaded json blobs parsed in parallel). Or in the other direction, I suppose, if they have a lot of cores and bandwidth and _really_ want the SQS queue to move fast.

Reply:
Thanks, let's add a little bit more context around this. 

I think this changelog is a bit too modest, it should specifically mention the performance improvement and that tuning max_number_of_messages should not be necessary anymore.

You should also mention the small object use case and the potential to increase memory usage.

The changelogs here are usually terse but that isn't actually a requirement. You could add an entire sub-section if you wanted to.

Reply:
Done

Reply:
Looks great, thanks!